processor/transformprocessor/factory_test.go

```diff
@@ -753,3 +779,237 @@ func TestFactoryCreateSpanEventProcessor(t *testing.T) {
 		})
 	}
 }
+
+func createTestFuncFactory[K any](name string) ottl.Factory[K] {
+	type TestFuncArguments[K any] struct{}
+	createFunc := func(_ ottl.FunctionContext, _ ottl.Arguments) (ottl.ExprFunc[K], error) {
+		return func(_ context.Context, _ K) (any, error) {
+			return nil, nil
+		}, nil
+	}
+	return ottl.NewFactory(name, &TestFuncArguments[K]{}, createFunc)
+}
+
+func Test_FactoryWithFunctions_CreateTraces(t *testing.T) {
+	type testCase struct {
+		name                     string
+		statements               []common.ContextStatements
+		customSpanFunctions      []ottl.Factory[ottlspan.TransformContext]
+		customSpanEventFunctions []ottl.Factory[ottlspanevent.TransformContext]
+		wantErrorWith            string
+	}
+
+	tests := []testCase{
+		{
+			name: "with span functions : statement with added span func",
+			statements: []common.ContextStatements{
+				{
+					Context:    common.ContextID("span"),
+					Statements: []string{`set(cache["attr"], TestSpanFunc())`},
+				},
+			},
+			customSpanFunctions:      []ottl.Factory[ottlspan.TransformContext]{createTestFuncFactory[ottlspan.TransformContext]("TestSpanFunc")},
+			customSpanEventFunctions: []ottl.Factory[ottlspanevent.TransformContext]{},
+		},
+		{
+			name: "with span functions : statement with missing span func",
+			statements: []common.ContextStatements{
+				{
+					Context:    common.ContextID("span"),
+					Statements: []string{`set(cache["attr"], TestSpanFunc())`},
```

Comment:
Could you also have cases that show the default functions can be removed from the list?